rotkehlchen/tests/unit/test_cost_basis.py

```diff
@@ -790,132 +827,45 @@ def test_accounting_average_cost_basis(accountant):
-        timestamp=5,
-        rate=FVal(500),
-        index=5,
-    ))
-    assert asset_events.acquisitions_manager.remaining_amount == FVal(2)
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(475)
-
-    # see that using more than the available acquisitions adds a MissingAcquisition
-    assert asset_events.acquisitions_manager.calculate_spend_cost_basis(
-        spending_amount=FVal(3.5),
-        spending_asset=asset,
-        timestamp=0,
-        missing_acquisitions=cost_basis.missing_acquisitions,
-        used_acquisitions=asset_events.used_acquisitions,
-        settings=cost_basis.settings,
-        timestamp_to_date=cost_basis.timestamp_to_date,
-        average_cost_basis=asset_events.acquisitions_manager.current_average_cost_basis,
-    ).is_complete is False
-    assert asset_events.acquisitions_manager.remaining_amount == ZERO
-    # it is negative due to the missing acquisition.
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(-356.25)
-    assert cost_basis.missing_acquisitions == [
-        MissingAcquisition(
-            asset=asset,
-            time=0,
-            found_amount=FVal(2),
-            missing_amount=FVal(1.5),
-        ),
-    ]
+    manager = cost_basis.get_events(A_ETH).acquisitions_manager
+
+    # Check 2 simple cases - a spend with positive pnl and a spend with negative pnl
+    add_acquisition(pot, amount=FVal(2), price=FVal(10))  # Buy 2 ETH for $10  total acb: $20
+    current_total_acb = events[0].price * events[0].taxable_amount
+    current_amount = events[0].taxable_amount  # 2
+    assert manager.current_total_acb == current_total_acb == FVal(20)
+    add_spend(pot, amount=FVal(1), price=FVal(15))  # Sell 1 ETH for $15  pnl: 1 * (15 - 10) = $5
+    assert events[1].pnl.taxable == events[1].taxable_amount * (events[1].price - current_total_acb / current_amount) == FVal(5)  # noqa: E501
+    current_total_acb *= (current_amount - events[1].taxable_amount) / current_amount  # total acb: 20 * (2 - 1) / 2 = $10  # noqa: E501
+    current_amount -= events[1].taxable_amount  # 1
+    add_acquisition(pot, amount=FVal(1), price=FVal(30))  # Buy 1 ETH for $30  total acb: 10 + 1 * 30 = $40  # noqa: E501
+    current_total_acb += events[2].price * events[2].taxable_amount
+    current_amount += events[2].taxable_amount  # 2
+    assert manager.current_total_acb == current_total_acb == FVal(40)
+    add_spend(pot, amount=FVal(1.5), price=FVal(40))  # Sell 1.5 ETH for $40  pnl: 1.5 * (40 - 20) = $30  # noqa: E501
+    assert events[3].pnl.taxable == events[3].taxable_amount * (events[3].price - current_total_acb / current_amount) == FVal(30)  # noqa: E501
+    current_total_acb *= (current_amount - events[3].taxable_amount) / current_amount  # total acb: 40 * (2 - 1.5) / 2 = $10  # noqa: E501
+    current_amount -= events[3].taxable_amount  # 0.5
+
+    # Now check that if acquisitions don't change the cost basis, everything remains correct.
```

Comment:
I don't understand this comment. I see bit more complicated numbers in these steps but what do you mean by the comment?